### PR TITLE
Fixed price oracle

### PR DESCRIPTION
**Pull request by blimp666**
New contract for #51

### CODE REVIEW

**Comment by blimp666** [comment]
I'm not sure about ```disallowToken``` - do we need it, or should I remove this function?

File: contracts/fixed_price_oracle.vy [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    self.daiAddress = dai_address
+    self.daiOracleAddress = dai_oracle_address
+    self.allowedTokens[dai_address] = True
+    self.supportedTokens[0] = dai_address
+    self.owner = msg.sender
+    self.name = 'PriceOracle'
+    
+@private
+@constant
+def _normalized_token_price(token_address: address) -> uint256:
```
Comment (gonchs) [inline contracts/fixed_price_oracle.vy]: What's the motivation for moving this into a private function and not having this code only in the public function without the _ prefix?

**Comment by gonchs** [comment]
disallowToken is not needed, remove.

File: contracts/fixed_price_oracle.vy [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+MAX_DAI_PRICE: constant(uint256) = 1050000000000000000 # 1.05*10**18
+# DAI_ORACLE_DIVIDER = 10**18 / PRICE_MULTIPLIER
+DAI_ORACLE_DIVIDER: constant(uint256) = 10000000000 # 10**10
+
+TokenAddressUpdated: event({token_address: indexed(address), token_index: indexed(int128)})
+
+name: public(string[16])
+owner: public(address)
+supportedTokens: public(address[5])
+allowedTokens: public(map(address, bool))
```
Comment (gonchs) [inline contracts/fixed_price_oracle.vy]: Why do we need `allowedTokens` new variable? What's the benefit over continuing to use `supportedTokens`?

File: contracts/fixed_price_oracle.vy [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+daiOracleAddress: public(address)
+
+@public
+def __init__(dai_address: address, dai_oracle_address: address):
+    assert dai_address != ZERO_ADDRESS
+    assert dai_oracle_address != ZERO_ADDRESS
+
+    self.daiAddress = dai_address
+    self.daiOracleAddress = dai_oracle_address
+    self.allowedTokens[dai_address] = True
```
Comment (gonchs) [inline contracts/fixed_price_oracle.vy]: Let's remove `allowedTokens` and just check if the address is DAI -> daiPriceOracle, if not -> 1.00

File: contracts/fixed_price_oracle.vy (hunk 1 above)
Comment (blimp666) [reply contracts/fixed_price_oracle.vy]: The motivation is to make it compatible with vyper 0.1.0b12 and future versions. b12 doesn't allow local calls of public functions, but we call this function from ```poolSize```.

Comment (gonchs) [reply contracts/fixed_price_oracle.vy]: I see. For now, we don't need to support b12 and probably won't need to for a long time until we upgrade the contract. Let's remove the prefix.

File: contracts/fixed_price_oracle.vy (hunk 3 above)
Comment (blimp666) [reply contracts/fixed_price_oracle.vy]: Done

File: contracts/fixed_price_oracle.vy (hunk 1 above)
Comment (blimp666) [reply contracts/fixed_price_oracle.vy]: Done

File: tests/test_fixed_price_oracle.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+from decimal import (
+    Decimal,
+)
+
+from tests.constants import (
+    DEADLINE
+)
+
+DAI_PRICE = 1010000000000000000 #1.01*10**18
+
```
Comment (gonchs) [inline tests/test_fixed_price_oracle.py]: Can we add a test to make sure that `normalized_token_prices` and `poolSize` will fail if the DAI price is < 0.95 or > 1.05 ?

Comment (blimp666) [reply tests/test_fixed_price_oracle.py]: sure

**Pull request merged**